packages/app/src/client/views/ring-health/RingTable.tsx

```diff
@@ -0,0 +1,146 @@
+} from "@material-ui/core";
+import { Card } from "client/components/Card";
+import { format } from "date-fns";
+import SkeletonTableBody from "./SkeletonTableBody";
+import { useNotificationService } from "client/services/Notification";
+import { Button } from "client/components/Button";
+import { Box } from "client/components/Box";
+import TokenDialog from "./TokenDialog";
+import { useRouteMatch, useHistory } from "react-router-dom";
+
+type Shard = {
+  id: string;
+  state: string;
+  timestamp: string;
+  zone: string;
+  address: string;
+  tokens: number[];
+  registered_timestamp: string;
+};
+
+type Payload = {
+  shards: Array<Shard>;
+  now: string;
+};
+
+type Props = {
+  ringEndpoint: string;
+  baseUrl: string;
+};
+
+function getRandomInt() {
+  return Math.floor(Math.random() * Math.floor(100000));
+}
+
+const RingTable = ({ ringEndpoint, baseUrl }: Props) => {
+  const history = useHistory();
+  const {
+    registerNotification,
+    unregisterNotification
+  } = useNotificationService();
+  const [shards, setShards] = useState<Array<Shard>>();
+  useEffect(() => {
+    async function fetchShards() {
+      try {
+        const response = await axios.get<Payload>(ringEndpoint);
+        setShards(response.data.shards);
+      } catch (e) {
+        const messageId = `${getRandomInt()}`;
+        const newNotification = {
+          id: messageId,
+          state: "error" as const,
+          title: `Could not load table`,
+          information: e.message,
+          handleClose: () =>
+            unregisterNotification({
+              id: messageId,
+              title: "",
+              information: ""
+            })
+        };
+        registerNotification(newNotification);
+      }
+    }
+    fetchShards();
+  }, [ringEndpoint, registerNotification, unregisterNotification]);
+
+  const tokenShardIdToDisplay = useRouteMatch<{shardId: string}>(`${baseUrl}/:shardId/token`)?.params.shardId;
+  const tokensToDisplay = shards?.find(s => s.id === tokenShardIdToDisplay)?.tokens
+
+  return (
+    <>
+      <TokenDialog tokens={tokensToDisplay} onClose={() => history.push(baseUrl)} />
+      <TableContainer component={Card}>
+        <Table aria-label="tenants">
+          <TableHead>
+            <TableRow>
+              <TableCell>ID</TableCell>
+              <TableCell>State</TableCell>
+              <TableCell>last Heartbeat</TableCell>
+              <TableCell>Zone</TableCell>
+              <TableCell>Address</TableCell>
+              <TableCell>Tokens</TableCell>
+            </TableRow>
+          </TableHead>
+          {shards ? (
+            <TableBody>
+              {shards.map(shard => (
+                <TableRow key={shard.id}>
+                  <TableCell component="th" scope="row">
+                    {shard.id}
+                  </TableCell>
+                  <TableCell>{shard.state}</TableCell>
+                  <TableCell>
+                    {format(new Date(shard.timestamp), "Pppp")}
```

Comment:
For this one, can we use date-fns to show it as "<time> ago", e.g. "2 mins ago"?

Reply:
it's possible, but it means we need to start account for timezones (server vs client timezone), which can be a bit tricky in JS.